lib/darr.h

```diff
@@ -291,7 +291,7 @@ void *__darr_resize(void *a, uint count, size_t esize, struct memtype *mt);
  */
 #define darr_ensure_i_mt(A, I, MT)                                             \
 	({                                                                     \
-		assert((int)(I) >= 0 && (int)(I) <= INT_MAX);                  \
+		assert((int)(I) >= 0 && (uint)(I) <= INT_MAX);                 \
```

Comment:
`darr_lasti()` can return -1 (int), is this fine to cast uint?

Reply:
well the output from darr_lasti() on some array should not be blindly used to ensure that index exists on a new array which is what this condition would require. `darr_lasti(a) == -1` is really meant as a way to say there's no last index b/c this array has 0 length, so it doesn't make sense to say "assure there is no valid index" by passing -1 to `darr_ensure_i()` :)

Reply:
That said we could allow the condition by simply doing nothing if the user passes `-1` to `darr_ensure_i()`...